### DISCUSSION TITLE
Module name shadowing should be as invalid as variable shadowing.

### DISCUSSION THREAD

**Discussion by JalonSolov** [vlang/v#12701] [Ideas]
From thread on discord:

spytheman
 — 
11/06/2021
exactly
at least to me, it is very confusing when a module name is used for a local variable or a parameter too

JalonSolov
 — 
11/06/2021
My vote?  That should be invalid. 
Would have to allow for import gg as g, then make any other use of g illegal.
Perl is crazy with that...  you can have %var, @var, and $var all in the same file... and then $var{key}, but that means the %var hash, not the $var plain variable.
I think once a name is defined as a type of thing, it should be invalid to use the same name for something else. 
It also goes along with the "Simple, safe" ideas in V.  Hard to confuse what it is if it's only ever one type of thing.

Alex M
 — 
Yesterday at 12:03 PM
yeah it'd be consistent with V not allowing shadowed variables

**Comment by jdonnerstag** [comment]
I like V-lang's simplicity. Which is why I like the simple rule of "every variable, function, constant from another module **must be** prefixed like `mymodule.myvar`. `mymodule` might be an alias, but that's it. Source code is easy to read and it is obvious that something is from another module. Zero ambiguity for the user (or compiler or language server) what it is.

In my humble opinion [selective imports](https://github.com/vlang/v/blob/master/doc/docs.md#selective-imports) is the problem and should be removed from V-lang again.

**Comment by JalonSolov** [comment]
Selective imports may be _an_ issue, but they aren't _the_ issue.
```v
import time

now := time.now()

time := 1

println('$time $now')
```
this is creating a variable named `time` that matches the imported module name... and it works.  The output is
```
1 2021-11-12 16:29:07
```
